src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -375,219 +809,116 @@ public Table GetCountersTable(Transaction tx, CollectionName collection)
 
-                return value;
+            long value = 0;
+            for (var i = 0; i < existingCount; i++)
+            {
+                value += GetPartialValue(i, blob);
             }
+
+            return value;
         }
 
-        public IEnumerable<(string ChangeVector, long Value)> GetCounterValues(DocumentsOperationContext context, string docId, string counterName)
+        private static bool TryGetRawBlob(DocumentsOperationContext context, string docId, string counterName, out BlittableJsonReaderObject.RawBlob blob)
         {
+            blob = null;
             var table = new Table(CountersSchema, context.Transaction.InnerTransaction);
 
-            using (GetCounterPartialKey(context, docId, counterName, out var keyPrefix))
+            using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(context, docId, out Slice key, out _))
             {
-                foreach (var result in table.SeekByPrimaryKeyPrefix(keyPrefix, Slices.Empty, 0))
-                {
-                    (string, long) val = ExtractDbIdAndValue(result);
-                    yield return val;
-                }
+                if (table.ReadByKey(key, out var tvr) == false)
+                    return false;
+
+                var data = GetData(context, ref tvr);
+                if (data.TryGet(Values, out BlittableJsonReaderObject counters) == false ||
+                    counters.TryGetMember(counterName, out object counterValues) == false ||
+                    counterValues is LazyStringValue || // deleted
+                    !(counterValues is BlittableJsonReaderObject.RawBlob counterValuesAsRawBlob))
+                    return false;
+
+                blob = counterValuesAsRawBlob;
+                return true;
             }
         }
 
-        private static (string ChangeVector, long Value) ExtractDbIdAndValue((Slice Key, Table.TableValueHolder Value) result)
-        {
-            var counterKey = result.Value.Reader.Read((int)CountersTable.CounterKey, out var size);
-            Debug.Assert(size > DbIdAsBase64Size);
-            var pCounterDbValue = result.Value.Reader.Read((int)CountersTable.Value, out size);
-            Debug.Assert(size == sizeof(long));
-            var changeVector = result.Value.Reader.Read((int)CountersTable.ChangeVector, out size);
-
-            return (Encoding.UTF8.GetString(changeVector, size), *(long*)pCounterDbValue);
-        }
-
-        private static LazyStringValue ExtractCounterName(JsonOperationContext context, TableValueReader tvr)
-        {
-            return TableValueToId(context, (int)CountersTable.Name, ref tvr);
-
-        }
-
-        public ByteStringContext.InternalScope GetCounterKey(DocumentsOperationContext context, string documentId, string name, string changeVector, out Slice partialKeySlice)
+        public IEnumerable<(string ChangeVector, long Value)> GetCounterValues(DocumentsOperationContext context, string docId, string counterName)
```

Comment:
Check where this is used. This should only be returned as a debug info / specific call. 
Shouldn't be in general use